RecoLocalCalo/HGCalRecProducers/plugins/EERecHitFromSoA.cc

```diff
@@ -0,0 +1,42 @@
+#include "EERecHitFromSoA.h"
+
+EERecHitFromSoA::EERecHitFromSoA(const edm::ParameterSet& ps) {
+  recHitSoAToken_ = consumes<HGCRecHitSoA>(ps.getParameter<edm::InputTag>("EERecHitSoATok"));
+  recHitCollectionToken_ = produces<HGCeeRecHitCollection>(collectionName_);
+}
+
+EERecHitFromSoA::~EERecHitFromSoA() {}
+
+void EERecHitFromSoA::acquire(edm::Event const& event,
+                              edm::EventSetup const& setup,
+                              edm::WaitingTaskWithArenaHolder w) {
+  cms::cuda::ScopedContextAcquire ctx{event.streamID(), std::move(w)};
+  HGCRecHitSoA recHitsSoA = event.get(recHitSoAToken_);
```

Comment:
I think this is related with the conversation above, regarding using a ```ConstHGCRecHitSoA``` and an abstraction on top.
Here my goal was to simply copy the data in the event, use it (without modifying it!), and then discard it. That is why its scope is limited to ```acquire()```. Unfortunately I am not entirely sure with what you mean by "who".

Reply:
Right, this is connected to a few other comments.

> Unfortunately I am not entirely sure with what you mean by "who".

`HGCRecHitSoA` itself holds raw pointers, and given its default destructor, it does not own the memory pointed by those pointers. Therefore, something else must own that memory (i.e. being responsible for freeing it). It is not clear to me from the code which object or piece of code is that owner. Recommended pattern is to have the event products to own their memory.

> Here my goal was to simply copy the data in the event, use it (without modifying it!), and then discard it.

That's fine, except we want to avoid unnecessary copies, which is why Event products are accessed e.g. through const references. Even if in this specific case the copy itself is cheap, seeing Event products being copied is usually code smell. In this case it also helped to reveal how easy it would be get writable access to arrays that should be read-only at this point.